trail/upload/tests/test_processors.py

```diff
@@ -0,0 +1,166 @@
+import os
+import shutil
+import tempfile
+
+import yaml
+
+from django.test import TestCase
+
+from upload.process_uploads.upload_wrapper import TemporaryUploadedFileWrapper
+from upload.process_uploads.upload_processor import UploadProcessor
+from upload.process_uploads.fits_processor import FitsProcessor
+
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class MockTmpUploadedFile:
+    def __init__(self, fname, sourcePath=""):
+        self.name = fname
+        self._sourcePath = sourcePath
+        self.sourceFilePath = os.path.join(sourcePath, fname)
+
+    def read(self):
+        with open(self.sourceFilePath, "rb") as f:
+            return f.read()
+
+    def temporary_file_path(self):
+        return self.sourceFilePath
+
+
+class TemporaryUploadedFileWrapperTestCase(TestCase):
+    """Tests the TemporaryUploadedFileWrapper functionality. """
+    testDataDir = os.path.join(TESTDIR, "data")
```

Comment:
The image data has been cropped, the WCSs have been translated to fit the crop center and header keys adjusted appropriately. The total size of the dataset is 4MB, most of which in the DECam data (2MB), which could be much much smaller if I leave only a couple of cropped headers in it and make the rest a 1x1 array. If 4MB is too much we can move stuff out, otherwise I'm quite fine with it living here in this state. 

Perhaps if we get some kind of validation stuff going we will want actual full datasets at which point additional resources might be required.